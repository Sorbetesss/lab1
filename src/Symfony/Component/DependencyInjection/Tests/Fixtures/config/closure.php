<?php

namespace Symfony\Component\DependencyInjection\Loader\Configurator;

return new class() {
    public function __invoke(ContainerConfigurator $c)
    {
        $c->services()
            ->set('closure_property', 'stdClass')
                ->public()
                ->property('foo', closure(service('bar')))
            ->set('bar', 'stdClass')
            ->set('closure_of_service_closure', 'stdClass')
                ->public()
                ->args([closure(service_closure('bar2'))])
            ->set('bar2', 'stdClass');
    }
};
