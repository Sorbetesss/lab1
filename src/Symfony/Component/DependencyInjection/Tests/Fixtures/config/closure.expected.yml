
services:
    service_container:
        class: Symfony\Component\DependencyInjection\ContainerInterface
        public: true
        synthetic: true
    closure_property:
        class: stdClass
        public: true
        properties: { foo: !service { class: Closure, arguments: [!service { class: stdClass }], factory: [Closure, fromCallable] } }
    closure_of_service_closure:
        class: stdClass
        public: true
        arguments: [!service { class: Closure, arguments: [!service_closure '@bar2'], factory: [Closure, fromCallable] }]
    bar2:
        class: stdClass
